mmdet/datasets/coco_with_text.py

```diff
@@ -310,17 +328,47 @@ def evaluate(self,
             )
-            # self._dump_predictions(filtered_predictions, metric_params['dataset'])
-            recall, precision, hmean, _ = text_eval(
-                filtered_predictions, gt_annotations, score_thr,
-                show_recall_graph=False,
-                use_transcriptions=True,
-                word_spotting=metric.startswith('word_spotting'))
-
-            print(f'Text detection recall={recall} precision={precision} hmean={hmean}')
-            eval_results[metric + '/hmean'] = float(f'{hmean:.3f}')
-            eval_results[metric + '/precision'] = float(f'{precision:.3f}')
-            eval_results[metric + '/recall'] = float(f'{recall:.3f}')
+
+            if metric_params['det_thr_range'] is None:
+                # self._dump_predictions(filtered_predictions, metric_params['dataset'])
+                recall, precision, hmean, _ = text_eval(
+                    filtered_predictions, gt_annotations, score_thr,
+                    show_recall_graph=False,
+                    use_transcriptions=True,
+                    word_spotting=metric.startswith('word_spotting'))
+
+                print(f'Text detection recall={recall} precision={precision} hmean={hmean}')
+                eval_results[metric + '/hmean'] = float(f'{hmean:.3f}')
+                eval_results[metric + '/precision'] = float(f'{precision:.3f}')
+                eval_results[metric + '/recall'] = float(f'{recall:.3f}')
+            else:
+                best_hmean = -1
+                best_det_thr = None
+                best_rec_thr = None
+                for det_thr in tqdm(metric_params['det_thr_range']):
+                    for rec_thr in metric_params['rec_thr_range']:
+                        filtered_predictions = self._filter_predictions(
+                            predictions, det_thr, rec_thr
+                        )
+                        recall, precision, hmean, _ = text_eval(
+                            filtered_predictions, gt_annotations, score_thr,
+                            show_recall_graph=False,
+                            use_transcriptions=True,
+                            word_spotting=metric.startswith('word_spotting'))
+
+                        if hmean >= best_hmean:
+                            best_hmean = hmean
+                            best_precision = precision
+                            best_recall = recall
+                            best_det_thr = det_thr
+                            best_rec_thr = rec_thr
+
+                eval_results[metric + '/hmean'] = float(f'{best_hmean:.3f}')
+                eval_results[metric + '/precision'] = float(f'{best_precision:.3f}')
+                eval_results[metric + '/recall'] = float(f'{best_recall:.3f}')
+
+                print(f'Text detection recall={best_recall} precision={best_precision} hmean={best_hmean} @ '
+                      f'det_thr={best_det_thr} and rec_thr={best_rec_thr}')
```

Comment:
Would it make sense to save thresholds into `eval_results` as well?

Reply:
fixed